include/regs.h

```diff
@@ -199,4 +199,141 @@
+    /* 13 */ HREG_MODE_SCHED, // toggle various vi special features
+    /* 14 */ HREG_MODE_UNUSED_14,
+    /* 15 */ HREG_MODE_PRERENDER,
+    /* 16 */ HREG_MODE_SETUP_FRAME, // set the base screen color (and by extension, shrink window color too)
+    /* 17 */ HREG_MODE_SCENE_CONFIG, // toggle drawing the scene config
+    /* 18 */ HREG_MODE_UNUSED_18,
+    /* 19 */ HREG_MODE_LETTERBOX, // toggle logging and view values for the letterbox system
+    /* 20 */ HREG_MODE_HEAP_FREE_BLOCK_TEST, // enable or disable heap free block test
+    /* 21 */ HREG_MODE_VI
+} HRegMode;
+
+// HREG_MODE_UCODE_DISAS
+#define R_UCODE_DISAS_TOGGLE      HREG(81) // < 0 enables and prints some hardware reg info for 1 frame, > 0 enables constant disas
+#define R_UCODE_DISAS_LOG_MODE    HREG(82) // 1 and 2 print counts, 3 enables fault client, 4 disables open/close disps
+#define R_UCODE_DISAS_LOG_LEVEL   HREG(83) // enables various logging within the dissasembler itself
+#define R_UCODE_DISAS_TOTAL_COUNT HREG(84) // read-only
+#define R_UCODE_DISAS_VTX_COUNT   HREG(85) // read-only
+#define R_UCODE_DISAS_SPVTX_COUNT HREG(86) // read-only
+#define R_UCODE_DISAS_TRI1_COUNT  HREG(87) // read-only
+#define R_UCODE_DISAS_TRI2_COUNT  HREG(88) // read-only
+#define R_UCODE_DISAS_QUAD_COUNT  HREG(89) // read-only
+#define R_UCODE_DISAS_LINE_COUNT  HREG(90) // read-only
+#define R_UCODE_DISAS_SYNC_ERROR_COUNT  HREG(91) // read-only
+#define R_UCODE_DISAS_LOAD_COUNT  HREG(92) // read-only
+#define R_UCODE_DISAS_DL_COUNT    HREG(93) // read-only
+
+// HREG_MODE_PRINT_MEMORY
+#define R_PRINT_MEMORY_INIT     HREG(94) // set to `HREG_MODE_PRINT_MEMORY` when init is complete
+#define R_PRINT_MEMORY_TRIGGER  HREG(81) // set to a negative number to print memory
+#define R_PRINT_MEMORY_ADDR     HREG(82) // middle 16 bits of the address to print from (80XXXX00)
+#define R_PRINT_MEMORY_SIZE     HREG(83) // value * 0x10 = length of the log
+
+// HREG_MODE_PLAY
+#define R_PLAY_INIT                   HREG(94) // set to `HREG_MODE_PLAY` when init is complete
+#define R_PLAY_RUN_UPDATE             HREG(81)
+#define R_PLAY_RUN_DRAW               HREG(82)
+#define R_PLAY_DRAW_SKYBOX            HREG(83)
+#define R_PLAY_DRAW_ROOM_FLAGS        HREG(84) // also enables drawing the scene config and room bg cover
+#define R_PLAY_DRAW_ACTORS            HREG(85)
+#define R_PLAY_DRAW_LENS_FLARES       HREG(86)
+#define R_PLAY_DRAW_SCREEN_FILLS      HREG(87)
+#define R_PLAY_DRAW_SANDSTORM         HREG(88)
+#define R_PLAY_DRAW_OVERLAY_ELEMENTS  HREG(89)
+#define R_PLAY_DRAW_ASSORTED_FLAGS    HREG(90) // see flags below
+#define R_PLAY_DRAW_SKYBOX_FILTERS    (R_PLAY_DRAW_ASSORTED_FLAGS & (1 << 0))
+#define R_PLAY_DRAW_SUN_AND_MOON      (R_PLAY_DRAW_ASSORTED_FLAGS & (1 << 1))
+#define R_PLAY_DRAW_LIGHTNING         (R_PLAY_DRAW_ASSORTED_FLAGS & (1 << 2))
+#define R_PLAY_DRAW_LIGHTS            (R_PLAY_DRAW_ASSORTED_FLAGS & (1 << 3))
```

Comment:
Maybe `R_PLAY_DRAW_ASSORTED_FLAGS` could be called `R_PLAY_DRAW_ENV_FLAGS`

I feel like the `R_` prefix should be reserved for regs, so maybe there could be defines for each flag like `#define PLAY_DRAW_ENV_SKYBOX_FILTERS (1 << 0)` instead?

Reply:
my main reasoning for avoiding ENV is cause lights are technically not a part of the environment system. Although they are closely tied (environment does the high level stuff and passes it on to low level lights), they are separate. Additionally, skybox is separate. Environment only really deals with loading the right files based on time of day/setting.

im ok with blurring the line a little bit here, but am not super enthusiastic about it

Reply:
Yeah I'm not as picky about naming regs tbh, in particular debug ones
I'd be okay with env_light or env_and_light if you prefer, or we can keep assorted even though slightly-inaccurate sounds better to me than generic

Reply:
I went with `PLAY_ENV_SKYBOX_FILTERS_FLAG` at first, but ive changed it to `PLAY_DRAW_ENV_SKYBOX_FILTERS` like was initially suggested. It reads more naturally I think to not redundantly state that its a flag, and it fits better with how the room draw flags are currently named

Reply:
Note: `PLAY_ENV_DRAW_SKYBOX_FILTERS` vs `PLAY_DRAW_ENV_SKYBOX_FILTERS`
I'm fine with either, but your comment says the latter, PR has the former